tools/clean

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+#
+# A basic script to clean all of the build artifacts and installed
+# modules.  Order matters, and it's easy to overlook something. In the
+# future this could be structured to be done via a single
+# one-build-tool-to-rule-them-all command, but currently we've a
+# plethora of tooling and this is at least a starting point.
+
+MAG="\033[95m"
+END="\033[0m"
+
+status() {
```

Comment:
We should include these functions into a `logging.sh` script, and then import them via `source logging.sh`.

Reply:
I think we want to move all of this stuff into something else, two main thoughts:
- when script logic gets to where we're at I'm inclined to move to something else like Python, Ruby, etc.
- more pragmatically, we should probably move this stuff to a Node-driven script (or into sigh) so that we gain cross-platform logic more easily (ex. Windows).

Reply:
Also this is very stopgap, we have meta-toolchain issues to consider, what with npm, sigh, and Bazel.

Reply:
Just my 2¢: re-using shell functions would make maintenance of the shells easier in the meantime. It's quicker than porting to python. And it's portable, given a windows env with bash installed.

Reply:
Yeah, I'm doing it... but I don't know that we should rely on Windows with bash. For further discussion but I am doing the logging.sh thing shortly.

Reply:
I agree with both of you. logging.sh sounds good for right now, and moving to JS or python sounds good in the slightly longer term.